### ISSUE TITLE
Improve Committee Set tests

### ISSUE THREAD

**Issue by IronGauntlets** [autonity/autonity#525]
# Problem

The tests defined under `committee_set_test.go` need to be improved based on the unresolved comments in pull request #512.

# Implementation

Majority of the test code is very much the same as the implementation code, therefore make them brittle when future changes are made to `committee_set.go`. We need to capture the behaviour of the committee set independent of the implementation such that small changes do not break the tests.

# Acceptance Criteria

Review from code owners.

# Notes

Should be completed after #512 has been merged into develop

**Comment by piersy** [comment]
The committee set implementation changed with #559 we should wait till #559 is merged before starting work on this ticket.
